server/src/main/java/org/elasticsearch/cluster/coordination/Join.java

```diff
@@ -65,53 +53,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeLong(lastAcceptedVersion);
     }
 
-    public DiscoveryNode getVotingNode() {
-        return votingNode;
-    }
-
-    public DiscoveryNode getMasterCandidateNode() {
-        return masterCandidateNode;
-    }
-
-    /**
-     * Temporary compatibility with serverless code repository.
-     */
-    public DiscoveryNode getSourceNode() {
-        return masterCandidateNode;
-    }
-
     public boolean masterCandidateMatches(DiscoveryNode matchingNode) {
         return masterCandidateNode.getId().equals(matchingNode.getId());
     }
 
-    public long getLastAcceptedVersion() {
-        return lastAcceptedVersion;
-    }
-
-    public long getTerm() {
-        return term;
-    }
-
-    public long getLastAcceptedTerm() {
-        return lastAcceptedTerm;
-    }
-
-    @Override
-    public String toString() {
-        return "Join{"
-            + "term="
-            + term
-            + ", lastAcceptedTerm="
-            + lastAcceptedTerm
-            + ", lastAcceptedVersion="
-            + lastAcceptedVersion
-            + ", votingNode="
-            + votingNode
-            + ", masterCandidateNode="
-            + masterCandidateNode
-            + '}';
-    }
-
     @Override
```

Comment:
Records generate reasonable equals and hashCode over all fields, so below could be removed

Reply:
Done, thanks